mqtt/mqtt-bridge/src/pump/builder.rs

```diff
@@ -128,10 +129,14 @@ where
         let topic_filters = make_topics(&self.remote.rules)?;
         let remote_topic_mappers_updates = TopicMapperUpdates::new(topic_filters);
 
+        let (retry_send, retry_recv) = mpsc::unbounded_channel();
+
         let rpc_subscriptions = RpcSubscriptions::default();
         let rpc = RemoteRpcMqttEventHandler::new(rpc_subscriptions.clone(), local_pump.handle());
-        let messages =
+        let mut messages =
             StoreMqttEventHandler::new(local_store, remote_topic_mappers_updates.clone());
+        messages.set_retry_sub_sender(retry_send);
```

Comment:
nit: is it possible to just add retry_sender to ctor?

Reply:
not every pump needs to retry subscriptions thus `retry_send` can be optional. not sure if optional argument is good for ctor

Reply:
I see. The ideal solution would be builder :) but maybe overkill here. So i'm fine with "setter"

Reply:
There is a macro trait for builder pattern :) https://github.com/colin-kiegel/rust-derive-builder

Reply:
well, some time ago we decided not to use too many derive macros: it makes code more difficult to read and navigate and it is super easy to implement it manually.

we write code once and read it many times...